File: src/main/java/gestion_restaurant/entity/MenuBurger.java
package gestion_restaurant.entity;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class MenuBurger {
    private Long id;
    private Long menuId;
    private Long burgerId;
    private Integer quantite = 1;
    private Menu menu;
    private Burger burger;
    public void setMenu(Menu menu) {
        this.menu = menu;
        this.menuId = (menu != null ? menu.getId() : null);
    }
    public void setBurger(Burger burger) {
        this.burger = burger;
        this.burgerId = (burger != null ? burger.getId() : null);
    }
}

File: Models/MenuBurger.cs
namespace gestion_restaurant.Models
{
    public class MenuBurger
    {
        public long Id { get; set; }

        public long MenuId { get; set; }
        public Menu Menu { get; set; }

        public long BurgerId { get; set; }
        public Burger Burger { get; set; }

        public int Quantite { get; set; }
    }
}
